src/components/form/partials/form-field/index.jsx

```diff
@@ -0,0 +1,62 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+
+const noop = () => {}
+
+class FormField extends React.PureComponent {
+    constructor(props){
+        super(props)
+        this.update = this.update.bind(this)
+    }
+
+    update(value) {
+        const {
+            name,
+            validate,
+            onUpdate,
+            onValidate
+        } = this.props
+
+        const error = validate({name, value})
+
+        onValidate({name, error})
+        onUpdate({name, value})
+    }
+
+    render() {
+        const {
+            component: Component,
+            error,
+            validate,
+            onUpdate,
+            onValidate,
+            ...rest
+        } = this.props
+
+        return (
+            <Component
+                {...rest}
+                error={{message: error}}
+                onUpdate={this.update}
+            />
+        )
+    }
+}
+
+FormField.PropTypes = {
+    component: PropTypes.func.isRequired,
+    error: PropTypes.string,
```

Comment:
Validation is checked on every change of the input. If there are multiple errors the last error will be passed to the component. As errors become valid they pop off until there are no more errors. eg. password field with length and character requirements, the error character requirements error will display until it has been met and if the length has not been met it will display